src/js/components/List/List.js

```diff
@@ -640,9 +650,15 @@ const List = React.forwardRef(
                 let displayPinned;
                 if (isPinned) {
                   // Pinned icon and settings
-                  const Pin = theme.list.icons.pin;
+                  // If pinned is passed in dataTableColumns use Lock icon.
+                  const Icon = inDataTableColumns ? Lock : theme.list.icons.pin;
                   const pinSize = theme.list.item.pinned.icon.size;
                   const pinPad = theme.list.item.pinned.icon.pad;
+                  const iconColor = inDataTableColumns ? pinTextColor : null;
```

Comment:
```suggestion
                  const iconColor = inDataTableColumns ? pinTextColor : undefined;
```
Subtle nuance that might not have an effect on the behavior, but I think `undefined` is more fitting here. Also, even though we don't use Typescript, I know that they tend to lean away from `null` to favor `undefined`